src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -2152,6 +2143,73 @@ JNI_METHOD(jobject, computePaseVerifier)
     return nullptr;
 }
 
+JNI_METHOD(jobject, getICDClientInfo)(JNIEnv * env, jobject self, jlong handle, jint jFabricIndex)
+{
+    chip::DeviceLayer::StackLock lock;
+
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    jobject jInfo = nullptr;
+    chip::app::ICDClientInfo info;
+    chip::FabricIndex fabricIndex = static_cast<chip::FabricIndex>(jFabricIndex);
+
+    ChipLogProgress(Controller, "getICDClientInfo(%u) called", fabricIndex);
+    AndroidDeviceControllerWrapper * wrapper = AndroidDeviceControllerWrapper::FromJNIHandle(handle);
+    VerifyOrReturnValue(wrapper != nullptr, nullptr, ChipLogError(Controller, "wrapper is null"));
+
+    err = JniReferences::GetInstance().CreateArrayList(jInfo);
```

Comment:
could you move JniLocalReferenceManager manager(env); before CreateArrayList? for example, line 2151, so that manager could release the objects created from CreateArrayList as well, thanks

Reply:
After moving it above CreateArrayList, I get a "use of deleted local reference 0x71" Crash.